### DISCUSSION TITLE
Automate creation of metadata for youtube videos in tiddler fields.

### DISCUSSION THREAD

**Discussion by arunnbabu81** [buggyj/tiddlyclip-plugin#123] [Ideas]
<img width="1129" alt="image" src="https://github.com/buggyj/tiddlyclip-plugin/assets/67494083/affae54a-2d21-4eb2-8a19-5eba33e5cc1b">

https://www.youtube.com/watch?v=-CivY0YkrVU

Here is a example of youtube video with metadata. I want to add this metadata into tiddler fields. 

For example,
This is a sample metadata form this youtube video - `Song Title: Chola Chola - Tamil` I want this to be added into tiddler fields like this so that `fieldname` will be `Song Title` and `fieldvalue` will be `Chola Chola - Tamil`. Here the delimiter is `:`

Similarly I want to do for other metadata which I select in a single step.

Is it possible to automate it using tiddlyclip.

What I am planning is to multi-select the metadata which I need and snip it.

**Comment by buggyj** [comment]
what is the application - what type of tiddler?  i.e. is it part of a video tiddler or a normal tiddler?

**Reply by arunnbabu81** [reply]
> what is the application - what type of tiddler? i.e. is it part of a video tiddler or a normal tiddler?

It's a part of YouTube video tiddlers
These are rule used.
![Screenshot_20230606_143923_Chrome](https://github.com/buggyj/tiddlyclip-plugin/assets/67494083/0c596a04-b6ce-4b92-96e1-03cd59e0a1c2)



If it's not possible with video tiddler... I am ok with displaying it in the video tiddler using a ViewTemplate

**Reply by arunnbabu81** [reply]
@buggyj can anything be done in this regard?

**Reply by buggyj** [reply]
It is not possible to add multiply fields in the way you propose.
It will be possible to add then one at a time with the new version of the tiddlyclip-plugin

**Reply by arunnbabu81** [reply]
> It is not possible to add multiply fields in the way you propose. It will be possible to add then one at a time with the new version of the tiddlyclip-plugin

Thats ok. I am fine with it. Is the update to tiddlyclip plug in nearby ?

**Reply by buggyj** [reply]
I will prepare a pre-release soon

**Reply by arunnbabu81** [reply]
>I will prepare a pre-release soon

Is it available for testing ?

**Reply by buggyj** [reply]
There is a new pre-release version included in the demo for my new addon:

https://tiddlysidenote.tiddlyhost.com/

**Reply by arunnbabu81** [reply]
> It will be possible to add then one at a time with the new version of the tiddlyclip-plugin

how to do this?

**Reply by buggyj** [reply]
It is possible to index field names with an index using `@(#fieldname)` where `#fieldname` contains the name of a field. e.g.

```
#field=$text
@(#field)=some text...
```
is the same as

```
$text=some text..
```

I have used this to add fields to a tiddler, clip the text to want to use for the name of the field into @cptext (using the `Cptext` command) and then clip the text for the field content using the `createfield` command.

![image](https://github.com/buggyj/tiddlyclip-plugin/assets/2855795/0cab00ed-4562-4c94-91da-29950f60941c)


here are the tiddlyclip tiddlers

[fieldindex.json.zip](https://github.com/buggyj/tiddlyclip-plugin/files/12345123/fieldindex.json.zip)

**Reply by arunnbabu81** [reply]
is it similar to this method ? https://github.com/buggyj/tiddlyclip-plugin/discussions/72

I already have `AddField ` json in my wiki. Will it cause any conflict. I was not able to make it work in my testing. I will upload a demo soon.

https://tiddlyclip-test.tiddlyhost.com/

**Reply by buggyj** [reply]
the tiddler `withCptextDo` needs to be of type

`application/x-bclip`

**Comment by arunnbabu81** [comment]
> There is a new pre-release version included in the demo for my new addon:
> 
> https://tiddlysidenote.tiddlyhost.com/

Thank you @buggyj . I am having a look at this now. It looks interesting and will give my feedback as I use it more. What are the changes in the new tiddlyclip plug in?

**Reply by buggyj** [reply]
There are lots of changes. Unfortunately I have just caught covid (for the first time) and an feeling rather unwell! I will add some more details when I can.

**Reply by arunnbabu81** [reply]
> There are lots of changes. Unfortunately I have just caught covid (for the first time) and an feeling rather unwell! I will add some more details when I can.

Take care of your health @buggyj . Wishing you a speedy recovery.

**Reply by arunnbabu81** [reply]
> There are lots of changes. Unfortunately I have just caught covid (for the first time) and an feeling rather unwell! I will add some more details when I can.



What does this plug in do ? $:/plugins/bj/tcedittables

Is it always needed to be used with the new version of tiddlyclip plug in?

**Comment by buggyj** [comment]
This plugin is not required by tiddlyclip.

It provide view mode editing of tiddlyclip tables, via the context menu.


see  $:/bj/urlcardconfig

and $:/bj/tableconfig

**Comment by arunnbabu81** [comment]
@buggyj How to get clips like this tiddler - https://tiddlysidenote.tiddlyhost.com/#tiddlywiki.substack.com

**Reply by buggyj** [reply]
That is what I call a sidenote tiddler - the instructions to create sidenote tiddlers is given in that tiddlywiki

**Reply by buggyj** [reply]
that tiddler was created with tiddlysidenote

**Reply by buggyj** [reply]
you can see the rules for creating the sidenote tiddlers in the tiddlyclip config tiddler call `TiddlyClipSys` - see TNSput

**Reply by buggyj** [reply]
The purpose of TiddlySide Notes is to have a single 'bookmarks' tiddlywiki that is always connected to the browser addon, and when you navigate to a webpage, previous notes and tags of that webpage automatically appear in the sidebar. If a webpage has not been 'tiddly bookmarked' then a blue button is shown, otherwise it is red. So this makes tiddlywiki the storage for bookmarks and these can be searched and organised in the tw way.

**Reply by arunnbabu81** [reply]
Sorry for the delay in replying. I was busy for a few days following the birth of my second child. I am trying to find how this sidenote works. How to get the `config` for sidenote in the tiddlyclip context menu

**Reply by buggyj** [reply]
Congratulations! I am very happy for you.

I am not sure what you trying to do with tiddlyclip.

Tiddlyclip does not have a sidebar.

**Reply by arunnbabu81** [reply]
> I am not sure what you trying to do with tiddlyclip.

I guess I haven't fully understood the concept of sidenote yet. I will read the docs once more and return here

**Reply by arunnbabu81** [reply]
I think I have understood the concept. How to add contents to the sidenote from a webpage. Text can be added by `typing` directly, `drag and drop` text or `copy paste` text. Is there any way to add images (I meant something similar to pin and snap feature in tiddlyclip)

What do you do with the sidenote tiddlers later on,  if I want to move them into different wikis ? Use the usual TW way of drag and drop ?
What is the `tid` feature in the new tiddlyclip pre-release ? Can `tid` feature be used to transfer tiddlers to other wikis

**Reply by buggyj** [reply]
Sidenote tiddlers are ordinary tiddlers. The tid option has always existed. It copied tiddlers by title or tag

**Comment by arunnbabu81** [comment]
I tried installing the pre-release tiddlyclip plug in and now there are two `click to dock ` viewtoolbar buttons for each tiddler

<img width="408" alt="image" src="https://github.com/buggyj/tiddlyclip-plugin/assets/67494083/f38d1b23-d007-497b-ab92-3e3cd007ac8b">

On clicking dark one of them, I get this error message

<img width="563" alt="image" src="https://github.com/buggyj/tiddlyclip-plugin/assets/67494083/c644fdd0-b0e4-4e6e-ba54-62c187eb413a">

On clicking the light button, I get only snip option in the tiddlyclip context menu
<img width="634" alt="image" src="https://github.com/buggyj/tiddlyclip-plugin/assets/67494083/e3cf1686-d345-488c-84dc-916fa629c9dc">


I just thought of reporting this issue. Can the new tiddlyclip version be used in my daily wikis ?

**Reply by buggyj** [reply]
The light button is new to tiddlyclip, the dark one is the old one we created for your pdfs. Do you have an example wiki i can look at?

**Reply by arunnbabu81** [reply]
sorry for the delay @buggyj 
[Here is the demo wiki](https://tiddlyclip-test.tiddlyhost.com/)

**Reply by buggyj** [reply]
try this.
[localtcplugin(2).json.zip](https://github.com/buggyj/tiddlyclip-plugin/files/12262474/localtcplugin.2.json.zip)

**Reply by arunnbabu81** [reply]
> try this. [localtcplugin(2).json.zip](https://github.com/buggyj/tiddlyclip-plugin/files/12262474/localtcplugin.2.json.zip)

Now the error is gone, Thank you @buggyj

**Comment by arunnbabu81** [comment]
Can the tiddlysidenote load tiddler body contents in the sidebar if the webpage opened has the same `url` as the fieldvalue of the `location` field of a tiddler

**Reply by buggyj** [reply]
The sidenote addon sends the url to the tiddlyclip plugin, which then retrieves the tiddler with the title that matches the url, and sends the rendered contents of the text field back to the sidenote addon. As tiddler titles are unique, only one tiddler will match the url, this make the logic straight forwards, either a tiddler already exists with the url as title or can be created. I do not know what other data could be used to uniquely create a tiddler title if the url was put into a `location` field.

**Reply by arunnbabu81** [reply]
> The sidenote addon sends the url to the tiddlyclip plugin, which then retrieves the tiddler with the title that matches the url, and sends the rendered contents of the text field back to the sidenote addon. As tiddler titles are unique, only one tiddler will match the url, this make the logic straight forwards, either a tiddler already exists with the url as title or can be created. I do not know what other data could be used to uniquely create a tiddler title if the url was put into a `location` field.

Tiddlyclip clip contents from a webpage into a tiddler with `url` of the webpage being saved in the `location` field of that tiddler.

This is one idea which came to my mind - create a viewtoolbar button (similar to the clip dock button) which when clicked will open the webpage from which the tiddler was created (based on the `url` in the `location` field). Clicking this viewtoolbar button can also simultaneously transfer the title of the tiddler to the sidenote addon so that the addon can load the tiddler content in the sidebar.

Now we can clip more text or images into the sidebar loaded tiddler directly from the webpage

Make sure that this new viewtoolbar button is seen only in tiddlers with `location` field by using some list filter (to avoid cluttering of the viewtoolbar)

**Reply by buggyj** [reply]
unfortunately it is not possible to send data to the sidebar, the sidebar has to request the data (this happens automatically when the sidebar is displayed). This means that if you used the location field you would have to ensure that the url does not appear in more than one location field. (otherwise there would be 2 or more tiddlers with the same location field and an arbitrary choice would have to be made as to which one to use). I can't think how to ensure that the url appears only in one location field.

**Reply by buggyj** [reply]
The type of change you are suggesting is more of an extension to tiddlyclip. From the feedback I get it appears that tiddlyclip is too complicated for most uses, hence I made tiddly side note to be a simple as I could, separate from tiddlyclip.

**Reply by arunnbabu81** [reply]
> The type of change you are suggesting is more of an extension to tiddlyclip. 


Is such a change feasible with tiddlyclip. I meant the viewtoolbar button  to open the tiddler in sidebar of the clipped webpage 


> From the feedback I get it appears that tiddlyclip is too complicated for most uses, hence I made tiddly side note to be a simple as I could, separate from tiddlyclip.


I prefer tiddlyclip more than sidenote. It allows advanced customisation....

**Reply by buggyj** [reply]
I do not know if what you are suggesting could be done. Writing browser addon code is much more difficult than writing code for a tiddlywiki extension, it is very time consuming. I expect that to make such a change to tiddlyclip would take 2 weeks. Given that the chrome browser is stopping support for the old Addons (v2 interface), I would be faced with writing the same code twice!

**Reply by arunnbabu81** [reply]
> I do not know if what you are suggesting could be done. Writing browser addon code is much more difficult than writing code for a tiddlywiki extension, it is very time consuming. I expect that to make such a change to tiddlyclip would take 2 weeks. Given that the chrome browser is stopping support for the old Addons (v2 interface), I would be faced with writing the same code twice! 
> 

I don't think the change I asked for is an urgent one. Just that it might be a good addition to tiddlyclip. Consider about implementing this only when you have enough free time. 



I would be interested in knowing about the new changes you made in the new version of tiddlyclip. Documenting those changes should get more priority I guess.

**Reply by buggyj** [reply]
I use SideNote for bookmarks and creating a timestamped journal (journaling is enabled in the addons options) and tiddlyclip for themed aggregation of clips (I have many tiddlywikis). 

There are a number of enhancements to tiddlyclip, It is probably not worthwhile documenting all of them as most people don't read documentation, I may make some videos instead.
